### PR TITLE
Test/text area

### PR DESCRIPTION
**Pull request by pakoval**

### CODE REVIEW

File: src/components/__tests__/TextArea.spec.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    wrapper = shallowMount(TextArea, {
+      propsData: {
+        placeholder: "Type text..",
+      },
+    });
+    vm = wrapper.vm;
+  });
+
+  it("does a wrapper exist", () => {
+    expect(wrapper.isVueInstance).toBeTruthy();
```
Comment (silchencko) [inline src/components/__tests__/TextArea.spec.ts:18]: it's ok to play with this test several times, but I wouldn't test it in every component. It doesn't have a lot of value

File: src/components/__tests__/TextArea.spec.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  });
+
+  it("check class of TextArea", () => {
+    expect(wrapper.classes()).toContain("textarea");
+  });
+
+  it("check placeholder", () => {
+    expect(wrapper.attributes("placeholder")).toBe("Type text..");
```
Comment (silchencko) [inline src/components/__tests__/TextArea.spec.ts:26]: nice!

File: src/components/__tests__/TextArea.spec.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  });
+
+  it("should set isReadonly", async () => {
+    expect(vm.isReadonly).toBeFalsy();
```
Comment (silchencko) [inline src/components/__tests__/TextArea.spec.ts:30]:
```suggestion
    expect(
      wrapper.find("textarea").element.getAttribute("readonly")
    ).toBeFalsy();
```

Comment (silchencko) [reply src/components/__tests__/TextArea.spec.ts:30]: or `wrapper.element` as you use in other tests

this test was too simple
It sets the prop and check this prop 
let's rather test the final result of changing this prop

File: src/components/__tests__/TextArea.spec.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    await wrapper.setProps({ isReadonly: true });
+    expect(vm.isReadonly).toBeTruthy();
+  });
+
+  describe("textareaAdjust", () => {
+    const smallText = "hello";
+
+    it("should call textareaAdjust", async () => {
```
Comment (silchencko) [inline src/components/__tests__/TextArea.spec.ts]:
```suggestion
    it("should emit input event", async () => {
```

File: src/components/__tests__/TextArea.spec.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  });
+
+  it("check placeholder", () => {
+    expect(wrapper.attributes("placeholder")).toBe("Type text..");
+  });
+
+  it("should set isReadonly", async () => {
+    expect(vm.isReadonly).toBeFalsy();
+    await wrapper.setProps({ isReadonly: true });
+    expect(vm.isReadonly).toBeTruthy();
```
Comment (silchencko) [inline src/components/__tests__/TextArea.spec.ts]:
```suggestion
    expect(wrapper.find("textarea").element.getAttribute("readonly")).toBe(
      "readonly"
    );
```

File: src/components/__tests__/TextArea.spec.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+      const textareaEl = wrapper.element as HTMLTextAreaElement;
+      await wrapper.setValue(text);
+      expect(textareaEl.value).toEqual(text);
+      expect(textareaEl.style.fontSize).toBe("18px");
+
+      await wrapper.setValue(smallText);
+      expect(textareaEl.value).toEqual(smallText);
+      expect(textareaEl.style.fontSize).toBe("22px");
+    });
```
Comment (silchencko) [inline src/components/__tests__/TextArea.spec.ts:63]: nice!

File: src/components/__tests__/TextArea.spec.ts (hunk 5 above)
Comment (pakoval) [reply src/components/__tests__/TextArea.spec.ts]: yep it's better decision!

File: src/components/__tests__/TextArea.spec.ts (hunk 1 above)
Comment (pakoval) [reply src/components/__tests__/TextArea.spec.ts:18]: i tried a new for me property isVueInstance instead exists()

Comment (silchencko) [reply src/components/__tests__/TextArea.spec.ts:18]: oh, ok

**Review by silchencko — APPROVED** [review]

**Pull request merged**